model/storage/disk/disk_model_store.py

```diff
@@ -63,20 +67,42 @@ def delete_unreferenced_models(
 
             # If it is not in valid_hotkeys and out of grace period remove it.
             if hotkey not in valid_models_by_hotkey:
-                bt.logging.trace(
-                    f"Removing directory for unreferenced hotkey: {hotkey} if out of grace."
+                deleted_hotkey = utils.remove_dir_out_of_grace(
+                    hotkey_path, grace_period_seconds
                 )
-                utils.remove_dir_out_of_grace(hotkey_path, grace_period_seconds)
+                if deleted_hotkey:
+                    bt.logging.trace(
+                        f"Removed directory for unreferenced hotkey: {hotkey}."
+                    )
+
             else:
                 # Check all the model subfolder paths.
                 hotkey_dir = Path(hotkey_path)
                 model_subfolder_paths = [
                     str(d) for d in hotkey_dir.iterdir() if d.is_dir
                 ]
 
+                # Check all the snapshot subfolder paths
```

Comment:
Mind adding a quick comment here that describes the expected folder layout?

Reply:
Done.